auto-self-nominate/src/index.ts

```diff
@@ -133,14 +128,33 @@ function supplementaryDeposit(
     if (!info.hasOwnProperty(address)) {
         throw new Error(`SelfNominate first with specific deposit value before repeating`);
     }
-    const [deposit] = info[address];
-    if (deposit < targetDeposit) {
-        return targetDeposit - deposit;
+    const { deposits } = info[address];
+    if (deposits < targetDeposit) {
+        return targetDeposit - deposits;
     } else {
         return 0;
     }
 }
 
+async function checkStateAndSendAlarm(bestBlockNumber: number, address: string): Promise<boolean> {
+    const banned = (await getStatesInfos(sdk, "Banned", bestBlockNumber)) as BannedAccountsInfo;
+    const jailed = (await getStatesInfos(sdk, "Jailed", bestBlockNumber)) as JailedAccountsInfo;
+
+    if (banned.accounts.includes(address)) {
+        const content = `Your account ${address} is permanently banned`;
+        email.sendError(content);
+        slack.sendError(content);
+        return true;
+    } else if (jailed.hasOwnProperty(address)) {
```

Comment:
No need to use `else if` when the preceding block always returns.

Reply:
Fixed.